app/containers/ITunesContainer/AllTracksContainer/index.js

```diff
@@ -0,0 +1,190 @@
+    if (loaded) {
+      setLoading(false);
+    }
+  }, [iTunesData, iTunesError]);
+
+  useEffect(() => {
+    if (searchQuery && !iTunesData?.results?.length) {
+      dispatchITunesSongs(searchQuery);
+      setLoading(true);
+    }
+  }, []);
+
+  const handleOnChange = (sQuery) => {
+    if (!isEmpty(sQuery)) {
+      dispatchITunesSongs(sQuery);
+      setLoading(true);
+    } else {
+      dispatchClearITunesSongs();
+    }
+  };
+
+  const handleOnActionClick = (trackRef) => {
+    if (!isEmpty(currentSongRef) && trackRef !== currentSongRef) {
+      currentSongRef?.current?.pause();
+    }
+    setCurentSongRef(trackRef);
+  };
+
+  const debouncedHandleOnChange = debounce(handleOnChange, 200);
+
+  const renderTrackDetails = () => {
+    const items = get(iTunesData, 'results', []);
+    const totalCount = get(iTunesData, 'resultCount', 0);
+    return (
+      <If condition={!isEmpty(items) || loading}>
+        <CustomCard>
+          <Skeleton loading={loading} active>
+            <If condition={!isEmpty(searchQuery)}>
+              <div>
+                <T id="search_term" values={{ searchQuery }} />
+              </div>
+            </If>
+            <If condition={totalCount !== 0}>
+              <div>
+                <T id="matching_tracks" values={{ totalCount }} />
+              </div>
+            </If>
+            <For
+              of={items}
+              ParentComponent={Container}
+              renderItem={(item, index) => (
+                <Link data-testid="redirect" to={`/track/${item.trackId}`}>
+                  <ITunesCard key={index} onActionClick={handleOnActionClick} track={item} />
+                </Link>
+              )}
+            />
+          </Skeleton>
+        </CustomCard>
+      </If>
+    );
+  };
+  const renderErrorState = () => {
+    let iTuneError;
+    if (iTunesError) {
+      iTuneError = iTunesError;
+    } else if (isEmpty(searchQuery)) {
+      iTuneError = 'itune_search_default';
+    }
+    return (
+      !loading &&
+      iTuneError && (
+        <If condition={iTunesError} otherwise={<T data-testid="default-message" id={iTuneError} />}>
+          <T data-testid="error-message" text={iTunesError} />
+        </If>
+      )
+    );
+  };
+
+  return (
+    <CustomCard title={intl.formatMessage({ id: 'iTunes_search' })} maxwidth={maxwidth}>
+      <Search
+        data-testid="search-bar"
+        type="text"
+        placeholder={intl.formatMessage({ id: 'search_for_music' })}
+        onChange={(evt) => debouncedHandleOnChange(evt.target.value)}
+        onSearch={(searchText) => debouncedHandleOnChange(searchText)}
+      />
+      {renderTrackDetails()}
+      {renderErrorState()}
+    </CustomCard>
+  );
+}
+
+AllTracksContainer.propTypes = {
+  dispatchITunesSongs: PropTypes.func,
+  dispatchClearITunesSongs: PropTypes.func,
+  maxwidth: PropTypes.number,
+  iTunesData: PropTypes.shape({
+    resultCount: PropTypes.number,
+    results: PropTypes.array
```

Comment:
add in shape for this array as well please

Reply:
Will add the shape of results